python/flexflow/torch/model.py

```diff
@@ -1417,24 +1418,62 @@ def is_unsqueeze(slice_elem):
             """Returns if the slice is equivalent to unsqueezing that
             dimension."""
             return slice_elem is None
+
+        def is_truncate(slice_elem, old_size):
+            start = 0 if slice_elem.start == None else slice_elem.start
+            new_size = slice_elem.stop - start
+            return new_size < old_size
+        
+        def is_single_element(slice_elem):
+            return type(slice_elem) == int
+
         shape = tensor.dims
-        # Match dimensions from right to left
-        new_shape = []  # append then reverse
+
+        # Fewer slices than input dimensions
+        diff = len(shape) - len(slices)
+        if diff > 0:
+            slices_lst = list(slices)
+            for i in range(diff):
+                slices_lst.append(slice(None, None, None))
+            slices = tuple(slices_lst)
+
+        # Match dimensions from right to left                                                                  
+        new_shape = []  # append then reverse                                                                  
         j = len(shape) - 1
+        curr_tensor = copy.copy(tensor)
+
         for slice_elem in reversed(slices):
             if is_colon(slice_elem):
                 assert j >= 0
                 new_shape.append(shape[j])
                 j -= 1
             elif is_unsqueeze(slice_elem):
                 new_shape.append(1)
+            elif is_single_element(slice_elem):
+                assert j >= 0
+                splits = [1, shape[j] - 1]
+                curr_tensor = ffmodel.split(input=curr_tensor, sizes=splits, axis=j, name=name)[0]
+                new_shape.append(1)
+                j -= 1
+            elif is_truncate(slice_elem, shape[j]):
+                assert j >= 0
+                start = 0 if slice_elem.start == None else slice_elem.start
+                splits = []
+                # create splits
+                if start != 0: 
+                    splits.append(start) # truncation from left
+                splits.append(slice_elem.stop - start)
+                if slice_elem.stop < shape[j]: 
+                    splits.append(shape[j] - slice_elem.stop) # truncation from right
+                curr_tensor = ffmodel.split(input=curr_tensor, sizes=splits, axis=j, name=name)[0]
```

Comment:
In the case of truncation from the right is `[0]` correct, or would `[1]` be necessary?